File: src/backend/Plms.Api/Domain/Entities/Product.cs
namespace Plms.Api.Domain.Entities
{
    public class Product
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string Sku { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        
        public Guid? CategoryId { get; set; }
        public ProductCategory? Category { get; set; }

        public Guid? VendorId { get; set; }
        public Vendor? Vendor { get; set; }

        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: src/frontend/src/types/product.ts
export interface Vendor {
    id: string;
    code: string;
    name: string;
    isActive: boolean;
}

export interface ProductCategory {
    id: string;
    code: string;
    name: string;
    isActive: boolean;
}

export interface Product {
    id: string;
    sku: string;
    name: string;
    description: string;
    categoryId?: string;
    vendorId?: string;
    isActive: boolean;
    createdAt: string;
    updatedAt: string;
    categoryName?: string;
    vendorName?: string;
}

export interface CsvImportReport {
    totalRows: number;
    validRows: number;
    errorRows: number;
    errors: RowValidationError[];
}

export interface RowValidationError {
    rowNumber: number;
    sku: string;
    errorType: string;
    message: string;
}

export interface ImportSession {
    id: string;
    fileName: string;
    status: string;
    allowOverwrite: boolean;
    totalRows: number;
    validRows: number;
    errorRows: number;
    createdBy: string;
    createdAt: string;
    completedAt?: string;
    issues: RowValidationError[];
}

export interface ImportSessionSummary {
    id: string;
    fileName: string;
    status: string;
    allowOverwrite: boolean;
    totalRows: number;
    validRows: number;
    errorRows: number;
    createdBy: string;
    createdAt: string;
    completedAt?: string;
}

export interface ImportCommitResult {
    sessionId: string;
    importedCount: number;
    updatedCount: number;
    status: string;
}
